packages/traceability/issuer-api/migrations/1622458600677-RemoveTokenIds.ts

```diff
@@ -0,0 +1,119 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class RemoveTokenIds1622458600677 implements MigrationInterface {
+    name = 'RemoveTokenIds1622458600677';
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        /* 
+            CERTIFICATES
+        */
+        const oldCertificates = await queryRunner.query(
+            `SELECT id, "tokenId" FROM "issuer_certificate"`
+        );
+
+        await queryRunner.query(`ALTER TABLE "issuer_certificate" ALTER COLUMN "id" DROP DEFAULT`);
+        await queryRunner.query(`DROP SEQUENCE "issuer_certificate_id_seq"`);
+        await queryRunner.query(
+            `ALTER TABLE "issuer_certificate" DROP CONSTRAINT "UQ_6489c34207c69cdc7b90afb4491"`
+        );
+        await queryRunner.query(`ALTER TABLE "issuer_certificate" DROP COLUMN "tokenId"`);
+
+        await Promise.all(
+            oldCertificates.map(async (oldCertificate: any) => {
+                return queryRunner.query(
+                    `UPDATE "issuer_certificate" SET id = ${oldCertificate.tokenId} WHERE id = '${oldCertificate.id}'`
+                );
+            })
+        );
+
+        /*
+            CERTIFICATION REQUESTS
+        */
+        const oldCertificationRequests = await queryRunner.query(
+            `SELECT id, "requestId" FROM "issuer_certification_request"`
+        );
+
+        await queryRunner.query(
+            `ALTER TABLE "issuer_certification_request" DROP CONSTRAINT "UQ_551869cc9ee5caeccd53c966cdd"`
+        );
+        await queryRunner.query(
+            `ALTER TABLE "issuer_certification_request" DROP COLUMN "requestId"`
+        );
+
+        await queryRunner.query(
+            `ALTER TABLE "issuer_certification_request" RENAME COLUMN "issuedCertificateTokenId" to "issuedCertificateId"`
+        );
+
+        await queryRunner.query(
+            `ALTER TABLE "issuer_certification_request" ALTER COLUMN "id" DROP DEFAULT`
+        );
+        await queryRunner.query(`DROP SEQUENCE "issuer_certification_request_id_seq"`);
+
+        await Promise.all(
+            oldCertificationRequests.map(async (oldCertReq: any) => {
+                return queryRunner.query(
+                    `UPDATE "issuer_certification_request" SET id = ${oldCertReq.requestId} WHERE id = '${oldCertReq.id}'`
+                );
+            })
+        );
+    }
```

Comment:
Dangerous approach, we can lose all `tokenId`'s and down migration won't help.
Let's better first copy `tokenId`'s to 'id' and then drop `tokenId` column

Reply:
We should consider not dropping or rewriting columns at all, but just rename columns around and in future remove old ones. In this case this would rename `id` to `id_bak`, then rename `requestId` to `id`

Reply:
Good point @ioncreature. It's an edge case, since tokenIds are stored in memory, but a valid one.

@soanvig I would consider this only in cases where there's no clear `down()` migrations. Here it's unnecessary, as the `down()` migration will work seamlessly.